docs/dev.rst

```diff
@@ -156,37 +192,31 @@ You are then free to bind-mount any directory to any container. For instance, to
         volumes:
           - /path/to/edx-platform:/openedx/edx-platform
 
-This override file will be loaded when running any ``tutor dev ..`` command. The edx-platform repo mounted at the specified path will be automatically mounted inside all LMS and CMS containers. With this file, you should no longer specify the ``-v/--volume`` option from the command line with the ``run`` or ``runserver`` commands.
+This override file will be loaded when running any ``tutor dev ..`` command. The edx-platform repo mounted at the specified path will be automatically mounted inside all LMS and CMS containers. With this file, you should no longer specify the ``-m/--mount`` option from the command line.
 
 .. note::
-    The ``tutor local`` commands load the ``docker-compose.override.yml`` file from the ``$(tutor config printroot)/env/local/docker-compose.override.yml`` directory.
+    The ``tutor local`` commands load the ``docker-compose.override.yml`` file from the ``$(tutor config printroot)/env/local/docker-compose.override.yml`` directory. One-time jobs from initialisation commands load the ``local/docker-compose.jobs.override.yml`` and ``dev/docker-compose.jobs.override.yml``.
 
 Common tasks
 ------------
 
+.. _edx_platform_dev_env:
+
 Setting up a development environment for edx-platform
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Following the instructions :ref:`above <bind_mounts>` on how to bind-mount directories from the host above, you may mount your own `edx-platform <https://github.com/openedx/edx-platform/>`__ fork in your containers by running either::
-
-    # Mount from the volumes/ directory
-    tutor dev bindmount lms /openedx/edx-platform
-    tutor dev runserver --volume=/openedx/edx-platform lms
-
-    # Mount from an arbitrary directory
-    tutor dev runserver --volume=/path/to/edx-platform:/openedx/edx-platform lms
+Following the instructions :ref:`above <bind_mounts>` on how to bind-mount directories from the host above, you may mount your own `edx-platform <https://github.com/openedx/edx-platform/>`__ fork in your containers by running::
 
-    # Add your own volumes to $(tutor config printroot)/env/dev/docker-compose.override.yml
-    tutor dev runserver lms
+    tutor dev runserver --mount=/path/to/edx-platform lms
 
-If you choose any but the first solution above, you will have to make sure that your fork works with Tutor.
+But to achieve that, you will have to make sure that your fork works with Tutor.
 
-First of all, you should make sure that you are working off the ``open-release/maple.2`` tag. See the :ref:`fork edx-platform section <edx_platform_fork>` for more information.
+First of all, you should make sure that you are working off the ``open-release/maple.3`` tag. See the :ref:`fork edx-platform section <edx_platform_fork>` for more information.
 
 Then, you should run the following commands::
```

Comment:
Too bad users still need to run four extra commands after mounting edx-platform. I don't know if we can address this before the conference, but it's on my mind as something we could improve.

Just thinking out loud (not a change request for this PR): If we upstreamed the openedx-assets script into edx-platform behind `make static`, and then made Makefile targets for Python and JS dev requirements, this could become:
```
tutor dev run -m /path/to/edx-platform make requirements-dev static
```

Reply:
What would be a good command name for preparing the edx-platform repo? I'm thinking of creating a `tutor dev tasks` subcommand under which we would move `createuser` and `importdemocourse`:

    tutor dev tasks importdemocourse

Of course, the list of tasks would be the result of a filter, making it possible for plugins to add custom commands. We could then write something like:

    tutor dev tasks -m /path/to/edx-platform prepare-edx-platform

Reply:
@regisb I like the `tasks` subcommand idea! I wonder if it would be elegant to bring `init` (and `pre-init`) within the `tasks` abstraction as well . What would you think about:
* Making `tutor MODE init` an alias to `tutor MODE tasks init`
* Renaming the filter "commands:init" to "tasks:init", and namespacing new tasks under "tasks:*"
* Supporting `tasks --limit`, with the same semantics as `init --limit`

Anyway, in terms of `prepare-edx-platform`, what if we just called the task something generic like `prepare-repo`? That way, the discovery plugin could hook in commands to prepare a mounted course-discovery repo , and so on for other plugins.

Reply:
I'm not sure whether replacing the init commands by a single task would work -- at least with what I had in mind for the `tasks` subcommand. In my understanding, a "task" would be a command to be run in a single container. More importantly, subsequent command line arguments passed to `tutor local tasks ...` would be passed along to the script. For instance, this would allow the `createuser` script to support `--superuser --staff --password=...` arguments. We could have a single task trigger multiple commands in different containers (such as `init`) but I'm not sure whether this approach would play well with command line arguments.

Reply:
Okay, got it, I was imaging something more involved. I misread:
> the list of tasks would be the result of a filter, making it possible for plugins to add custom commands.

as something like:
> the list of commands run by a task would be the result of a filter, making it possible for plugins to add custom commands [to tasks]

What advantage would `tutor dev tasks` have over just putting scripts in the /openedx/bin folder (like `openedx-assets`), and then invoking them with `tutor dev run`? Is it that plugins can't currently add to /openedx/bin for existing images?

Reply:
> Is it that plugins can't currently add to /openedx/bin for existing images?

Yes, exactly. Adding stuff there requires rebuilding the image, which is inconvenient. We could also bind-mount a script folder, but this is inconvenient on Kubernetes.

Reply:
Got it. I'm in favor, then!

In the long term, I'd prefer if we could upstream any generally-useful scripts into the app repos instead of writing them into plugins as tasks, but I know that's often easier said than done...